### PR TITLE
Add Mattermost notifier

### PR DESCRIPTION
**Pull request by frezbo**
Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: pkg/notify/mattermost.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// *mattermost.Client4 and the mattermost direct message channel ID
+func NewMattermost(url, token, username string) (Notifier, error) {
+	client := mattermost.NewAPIv4Client(url)
+	client.AuthToken = token
+	client.AuthType = mattermost.HEADER_AUTH
+	client.HttpHeader = map[string]string{
+		"Authorization": fmt.Sprintf("Bearer %s", token),
+	}
+	botUser, res := client.GetMe("")
+	if res.StatusCode != 200 {
```
Comment (sohm22) [inline pkg/notify/mattermost.go]: Please use HTTP status code constants defined in net/http package

Comment (frezbo) [reply pkg/notify/mattermost.go]: Thanks for that :hugs: , I forgot about them, fixed :muscle:

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,8 @@ go 1.16
 
 require (
 	github.com/go-telegram-bot-api/telegram-bot-api v4.6.4+incompatible
-	github.com/pkg/errors v0.8.1
+	github.com/mattermost/mattermost-server/v5 v5.35.3
+	github.com/pkg/errors v0.9.1
```
Comment (PrasadG193) [inline go.mod:8]: Could you please run `go mod tidy`. I see redundant entries in go.sum

**Review by PrasadG193 — APPROVED** [review]
Overall looks good. Please run `go mod tidy` and we can merge after that

File: go.mod (hunk 2 above)
Comment (frezbo) [reply go.mod:8]:
```
covaccine-notifier on  feature/mattermost-notifier via 🐳 rootless 
❯ go mod tidy
covaccine-notifier on  feature/mattermost-notifier via 🐳 rootless 
❯ git diff
covaccine-notifier on  feature/mattermost-notifier via 🐳 rootless 
❯ 
```

I've already run `go mod tidy`. It doesn't seem to do anything. Am I missing something?

Comment (PrasadG193) [reply go.mod:8]: Oh ok. I saw a lot of changes in `go.sum` so was wondering if we forgot to do so. Thanks, let's merge this :rocket:

**Comment by frezbo** [comment]
I checked the mattermost code, the redundant go.sum deps are coming from the mattermost codebase itself, seems their deps are not so clean. Can't help much on that :sweat_smile: Thanks :hugs:

**Pull request merged**